plugins/aws/iam/trustedCrossAccountRoles.js

```diff
@@ -0,0 +1,96 @@
+            default: ''
+        }
+    },
+
+    run: function(cache, settings, callback) {
+        var whitelisted_accounts = settings.whitelisted_accounts || this.settings.whitelisted_accounts.default;
+        whitelisted_accounts = whitelisted_accounts.split(',');
+        // RegExp to get account ID from 'arn:aws:iam::{account ID}:root' principal
+        var accountRegExp = /(?<=arn:aws:iam::)(.*?)(?=:)/g;
+        var results = [];
+        var source = {};
+        
+        var region = helpers.defaultRegion(settings);
+
+        var listRoles = helpers.addSource(cache, source,
+            ['iam', 'listRoles', region]);
+
+        if (!listRoles) return callback(null, results, source);
+
+        if (listRoles.err || !listRoles.data) {
+            helpers.addResult(results, 3,
+                'Unable to query for IAM roles: ' + helpers.addError(listRoles));
+            return callback(null, results, source);
+        }
+
+        if (!listRoles.data.length) {
+            helpers.addResult(results, 0, 'No IAM roles found');
+            return callback(null, results, source);
+        }
+
+        var awsRolesFound = false;
+
+        listRoles.data.forEach(role => {
+            if (!role.AssumeRolePolicyDocument) return;
+
+            try {
+                var assumeRolePolicy = JSON.parse(decodeURIComponent(role.AssumeRolePolicyDocument));
+            } catch (e) {
+                helpers.addResult(results, 3,
+                    'IAM role policy document is not valid JSON.',
+                    'global', role.Arn);
+                return;
+            }
+            var restrictedAccounts = [];
+            var crossAccountRole = false;
+            
+            if (assumeRolePolicy.Statement && assumeRolePolicy.Statement.length) {
+                for (var s in assumeRolePolicy.Statement) {
+                    var statement = assumeRolePolicy.Statement[s];
+                    
+                    if (statement.Principal &&
+                        statement.Principal.AWS) {
+                        crossAccountRole = true;
+                        awsRolesFound = true;
+
+                        var account = statement.Principal.AWS.match(accountRegExp).toString();
+
+                        if (!whitelisted_accounts.includes(account)) restrictedAccounts.push(account);
+                    }
+                }
+
+                if (crossAccountRole && !restrictedAccounts.length) {
+                    helpers.addResult(results, 0,
+                        'Cross-account role :' + role.RoleName + ': contains trusted account ID(s)',
+                        'global', role.Arn);
+                }
+                else if (crossAccountRole) {
+                    helpers.addResult(results, 2,
+                        'Cross-account role :' + role.RoleName + ': contains these untrusted account ID(s): ' + restrictedAccounts.join(', '),
+                        'global', role.Arn);
+                }
+            }
```

Comment:
Passing condition is needed here for `IAM role does not contain trust relationship statements`.

Reply:
Again, the result will only be displayed if the role is cross-account (having cross-account principal) and it has trust relationship statements. If we don't find any such cross-account, a result is being displayed at the end that no cross-account role found.